### PR TITLE
[CALCITE-5344] Migrate Travis CI and AppVeyor configuration to Github Actions

### PR DESCRIPTION
**Pull request by F21**
The ASF is discontinuing Travis CI for testing, and it will no longer be available after 31 December 2022.

### CODE REVIEW

File: site/develop/index.md [hunk 1]
```diff
@@ -309,10 +309,9 @@ so it is better to stick with `org.checkerframework.checker.nullness.qual.Nullab
 
 ## Continuous Integration Testing
 
-Calcite exploits [GitHub actions](https://github.com/apache/calcite/actions?query=branch%3Amain)
-and [Travis](https://app.travis-ci.com/github/apache/calcite) for continuous integration testing.
-In the past, there were also Jenkins jobs on the [ASF-hosted](https://builds.apache.org/)
-infrastructure, but they are not maintained anymore.
+Calcite exploits [GitHub actions](https://github.com/apache/calcite/actions?query=branch%3Amain) for continuous
+integration testing. In the past, there were also Jenkins jobs on the [ASF-hosted](https://builds.apache.org/)
+infrastructure and [Travis](https://app.travis-ci.com/github/apache/calcite), but they are not maintained anymore.
```
Comment (asolimando) [inline site/develop/index.md:314]: I am not sure we want to keep this historical information around, but if we do we should also mention `AppVeyor` IMO

**Review by asolimando — APPROVED** [review]
LGTM, only one minor comment, I can't stress enough how this PR is welcome, thanks @F21!

File: site/develop/index.md (hunk 1 above)
Comment (F21) [reply site/develop/index.md:314]: Oh yeah, totally forgot about adding AppVeyor to it after migrating it to GH Actions. I suppose we can just remove that paragraph as it doesn't seem relevant to us.

Comment (asolimando) [reply site/develop/index.md:314]: +1 on the removal, we have git history if needed, but I guess nobody will really need this info

**Comment by F21** [comment]
Let's ship this! 🚀

**Pull request merged**